src/app/content/hooks/locationChange.ts

```diff
@@ -116,39 +116,58 @@ const resolvePage = async(
   }
 };
 
-const loadContentReference = (
-  book: Book,
-  bookPages: ReturnType<typeof flattenArchiveTree>,
-  reference: ReturnType<typeof getContentPageReferences>[0]
+const resolveExternalBookReference = async(
+  {archiveLoader, osWebLoader}: AppServices & MiddlewareAPI,
+  pageId: string
 ) => {
-  if (reference.bookUid || reference.bookVersion) {
-    throw new Error('BUG: Cross book references are not supported');
+  const bookId = (await archiveLoader.getBookIdsForPage(pageId)).filter((id) => BOOKS[id])[0];
+
+  if (!bookId) {
+    throw new Error(`BUG: ${pageId} could not be found in any configured books.`);
   }
-  if (!bookPages.find((search) => search.id === reference.pageUid)) {
-    throw new Error(`BUG: ${reference.pageUid} is not present in the ToC`);
+
+  const bookVersion = BOOKS[bookId].defaultVersion;
+
+  const osWebBook = await osWebLoader.getBookFromId(bookId);
+  const archiveBook = await archiveLoader.book(bookId, bookVersion).load();
+  const book = formatBookData(archiveBook, osWebBook);
+
+  if (!archiveTreeContainsSection(book.tree, pageId)) {
+    throw new Error(`BUG: archive thought ${book.id} would contain ${pageId}, but it didn't`);
```

Comment:
:+1: (just noting to self where the code sections are because there are so many tests in the diff 😄 )